WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -239,19 +231,57 @@ extension ProductFormViewModel {
 //
 extension ProductFormViewModel {
     func updateProductRemotely(onCompletion: @escaping (Result<EditableProductModel, ProductUpdateError>) -> Void) {
-        let updateProductAction = ProductAction.updateProduct(product: product.product) { [weak self] result in
-            switch result {
-            case .failure(let error):
-                ServiceLocator.analytics.track(.productDetailUpdateError, withError: error)
-                onCompletion(.failure(error))
-            case .success(let product):
-                ServiceLocator.analytics.track(.productDetailUpdateSuccess)
-                let model = EditableProductModel(product: product)
-                self?.resetProduct(model)
-                onCompletion(.success(model))
+        let remoteActionUseCase = ProductFormRemoteActionUseCase()
+        switch formType {
+        case .add:
+            remoteActionUseCase.addProduct(product: product, password: password) { [weak self] productResult, passwordResult in
+                switch productResult {
+                case .failure(let productError):
+                    onCompletion(.failure(productError))
+                case .success(let product):
+                    guard let passwordResult = passwordResult else {
+                        assertionFailure("Password result is expected to be non-nil when product result is successful.")
+                        onCompletion(.failure(.unknown))
+                        return
+                    }
+                    switch passwordResult {
+                    case .failure(let passwordError):
```

Comment:
I see a build warning here. 🤪

Reply:
I was originally going to pass the password error as an associated value, but I actually don't have a use for it yet. Updated in https://github.com/woocommerce/woocommerce-ios/pull/2785/commits/ec900ac19483a08c697256300524b4cf15b86bcf 🙂